### PR TITLE
Add cert-manager for API server

### LINKED ISSUE #58: Introduce cert-manager selfsigned for the API
We need to support TLS communication on the API. We will be using cert-manager and a self signed issuer for started, then we will refine the implementation to provide more options and to cover more components.

### PR DESCRIPTION
**Pull request by rh-hemartin**
* Add the cert-manager v1.15 version to helm crds
* Add a template file that generates a CA cert and a certificate for the API
* Change the API server to mount the certificate and use it
* Add instructions to get the CA certificate to query the API with curl

Closes #58

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -90,9 +90,13 @@ By default KubeArchive listens to `Event`s in the `test` namespace.
     ```bash
     kubectl port-forward -n kubearchive svc/kubearchive-api-server 8081:8081
     ```
+1. Get the Certificate Authority (CA) from the `kubearchive-api-server-tls` secret:
+    ```bash
+    kubectl get -n kubearchive secrets kubearchive-api-server-tls -o jsonpath='{.data.ca\.crt}' | base64 -d > ca.crt
```
Comment (mafh314) [inline DEVELOPMENT.md:95]: could you add '$' at the beginning of the command?

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -90,9 +90,13 @@ By default KubeArchive listens to `Event`s in the `test` namespace.
...
+    ```
 1. On a new terminal, use `curl` or your browser to perform a query:
     ```bash
-    curl localhost:8081/apis/apps/v1/deployments
+    curl --cacert ca.crt localhost:8081/apis/apps/v1/deployments
```
Comment (mafh314) [inline DEVELOPMENT.md:99]: could you add '$' at the beginning of the command?

File: DEVELOPMENT.md (hunk 1 above)
Comment (skoved) [reply DEVELOPMENT.md:95]: for code blocks like this, github will generate a button to copy all of the text. If the `$` is added at the beginning, you have to go to the beginning of the line to remove the `$` if you paste the command into your terminal which I think is kind of annoying

Comment (rh-hemartin) [reply DEVELOPMENT.md:95]: That's it, I'm keeping the commands without dollar sign so they are easier to copy.

**Review by mafh314 — APPROVED** [review]

**Pull request merged**